core/block_svg.js

```diff
@@ -272,18 +272,26 @@ Blockly.BlockSvg.prototype.getIcons = function() {
  * @param {Blockly.BlockSvg} newParent New parent block.
  */
 Blockly.BlockSvg.prototype.setParent = function(newParent) {
-  if (newParent == this.parentBlock_) {
+  if (newParent === this.parentBlock_) {
     return;
   }
   var svgRoot = this.getSvgRoot();
+  var oldXY = this.getRelativeToSurfaceXY();
```

Comment:
The ifs weren't an if/else, though.  It may still be right to pull this out, and to reorder the code in the way that you did, but it won't be doing the same thing as the old code.